src/pages/home/report/ReportActionCompose/SuggestionMention.tsx

```diff
@@ -279,8 +268,16 @@ function SuggestionMention(
     );
 
     useEffect(() => {
-        calculateMentionSuggestion(selection.end);
-    }, [selection, calculateMentionSuggestion]);
+        calculateMentionSuggestionRef.current = calculateMentionSuggestion;
+    }, [calculateMentionSuggestion]);
+
+    useEffect(() => {
+        if (!isComposerFocused) {
+            return;
+        }
+        calculateMentionSuggestionRef.current?.(selection.end);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isComposerFocused]);
```

Comment:
This change is to keep the behavior from your PR [here](https://github.com/Expensify/App/pull/31289).

Reply:
Good news that the iOS selection issue doesn't happen on the New Arch anymore.

The selection is still buggy though as you can see below,
<img width="239" alt="image" src="https://github.com/Expensify/App/assets/50919443/c2c21f40-363c-4472-ae59-115c18a95af5">

the selection is triggered thrice with the second one giving the selection based on the text length but I don't see the mention pop up.

Previously on the old arch, you would get
`1 @bernhard` even though the text is `A@bernhard`

So I removed the workaround.

But I still don't understand this one. What is broken?

Reply:
> This change is to keep the behavior from your PR [here](https://github.com/Expensify/App/pull/31289).

@bernhardoj I mean the suggestion menu doesn't open when revisit page.


https://github.com/Expensify/App/assets/117511920/9fe56d65-7fc9-4988-b28d-1fb87737eb09

Reply:
Ah I see, I didn't test the switching tab. But if I remove the code to recalculate when focused, the suggestion won't show anymore when refocusing. It does on `main`.

https://github.com/Expensify/App/assets/50919443/887a9e41-3bd6-4e30-94ce-dbf249b4ef7a

I guess it's fine to keep this one